contracts/TokenConversionLib.sol

```diff
@@ -0,0 +1,59 @@
+pragma solidity 0.4.23;
+
+import "./SafeMath.sol";
+
+library TokenConversionLib {
+    using SafeMath for uint256;
+
+    /**
+     *	@notice Calculate Utility token amount based on stake OST amount, conversion rate and conversion decimal
+     *
+     *	@param amountST amount of ST needs to be staked and converted to UT
+     *	@param conversionRate rate which is used to convert ST to UT
+     *	@param conversionRateDecimals represents number of decimal in conversion rate
+     *
+     *	@return  amountUT number of utility token which can be minted from given amountST
+     */
+
+    function calculateUTAmount(
+        uint256 amountST,
+        uint256 conversionRate,
+        uint8 conversionRateDecimals)
+        internal
+        pure
+        returns(
+        uint256 amountUT)
+    {
+        require(amountST > 0, 'Amount that needs to be converted should be greater than zero');
+        require(conversionRate > 0, 'Conversion Rate should be greater than zero');
+
+        amountUT = (amountST.mul(conversionRate)).div(10 ** uint256(conversionRateDecimals));
+        return amountUT;
+    }
+
+    /**
```

Comment:
I think this function should take `_amount` (as in what is sent to `requestStake`) and internally call `calculateUTAmount`, to save on calls from `Gateway`. Relatedly, I suggest renaming to something like `calculateMaxStake`.

In addition to any changes related to the above, the `@notice` should not refer to unstaking.

Reply:
1. As mentioned above, let discuss about this.  

2. Done 👍